### PR TITLE
rtl8812au: 5.6.4.2 -> 5.9.3.2

### PR DESCRIPTION
**Pull request by moni-dz**
###### Motivation for this change
ZHF: #122042 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 122188` at 2cdab46a run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages-libre.rtl8812au</li>
    <li>linuxPackages_hardkernel_4_14.rtl8812au</li>
    <li>linuxPackages_hardkernel_latest.rtl8812au</li>
    <li>linuxPackages_latest-libre.rtl8812au</li>
    <li>linuxPackages_testing_bcachefs.rtl8812au</li>
    <li>linuxPackages_xanmod.rtl8812au</li>
  </ul>
</details>
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/a7d1ab8d874eea87db9cfe9399e2833a">linuxPackages_4_4.rtl8812au</a></li>
    <li><a href="https://gist.github.com/7aa4cadec4318ae56fb101a176f346e8">linuxPackages_4_9.rtl8812au</a></li>
    <li><a href="https://gist.github.com/e7a03c6e2fd58763e27282160d60884c">linuxPackages_hardened.rtl8812au</a></li>
    <li><a href="https://gist.github.com/d97ad2e3da2a1111a3fad5f551de08fd">linuxPackages_latest_hardened.rtl8812au</a></li>
  </ul>
</details>
<details>
  <summary>7 packages built successfully:</summary>
  <ul>
    <li>linuxPackages.rtl8812au (linuxPackages_5_10.rtl8812au)</li>
    <li>linuxPackages_4_14.rtl8812au</li>
    <li>linuxPackages_4_19.rtl8812au</li>
    <li>linuxPackages_5_11.rtl8812au (linuxPackages_latest.rtl8812au)</li>
    <li>linuxPackages_5_4.rtl8812au</li>
    <li>linuxPackages_lqx.rtl8812au</li>
    <li>linuxPackages_zen.rtl8812au</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/os-specific/linux/rtl8812au/default.nix:46:5:
```
   |
46 |     license = licenses.gpl2;
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/name-and-version.md">name-and-version</a>

Did you mean to pass `pname` instead of `name` to `mkDerivation`?

Near pkgs/os-specific/linux/rtl8812au/default.nix:4:3:
```
  |
4 |   name = "rtl8812au-${kernel.version}-${version}";
  |   ^
```

Near pkgs/os-specific/linux/rtl8812au/default.nix:5:3:
```
  |
5 |   version = "5.9.3.2.20210427";
  |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by fortuneteller2k** [comment]
Result of `nixpkgs-review pr 122188` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages-libre.rtl8812au</li>
    <li>linuxPackages_hardkernel_4_14.rtl8812au</li>
    <li>linuxPackages_hardkernel_latest.rtl8812au</li>
    <li>linuxPackages_latest-libre.rtl8812au</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>linuxPackages_hardened.rtl8812au</li>
    <li>linuxPackages_latest_hardened.rtl8812au</li>
  </ul>
</details>
<details>
  <summary>11 packages built:</summary>
  <ul>
    <li>linuxPackages.rtl8812au (linuxPackages_5_10.rtl8812au)</li>
    <li>linuxPackages_4_14.rtl8812au</li>
    <li>linuxPackages_4_19.rtl8812au</li>
    <li>linuxPackages_4_4.rtl8812au</li>
    <li>linuxPackages_4_9.rtl8812au</li>
    <li>linuxPackages_5_11.rtl8812au (linuxPackages_latest.rtl8812au)</li>
    <li>linuxPackages_5_4.rtl8812au</li>
    <li>linuxPackages_lqx.rtl8812au</li>
    <li>linuxPackages_testing_bcachefs.rtl8812au</li>
    <li>linuxPackages_xanmod.rtl8812au</li>
    <li>linuxPackages_zen.rtl8812au</li>
  </ul>
</details>

File: pkgs/os-specific/linux/rtl8812au/default.nix [hunk 1]
```diff
@@ -42,8 +42,8 @@ stdenv.mkDerivation rec {
 
   meta = with lib; {
     description = "Driver for Realtek 802.11ac, rtl8812au, provides the 8812au mod";
-    homepage = "https://github.com/gordboy/rtl8812au-5.6.4.2";
-    license = licenses.gpl2;
+    homepage = "https://github.com/gordboy/rtl8812au-5.9.3.2";
+    license = licenses.gpl2Only;
     platforms = platforms.linux;
     maintainers = with maintainers; [ danielfullmer ];
```
Comment (jonringer) [inline pkgs/os-specific/linux/rtl8812au/default.nix:48]:
```suggestion
    maintainers = with maintainers; [ danielfullmer ];
    broken = kernel.kernelOlder "4.10" || kernel.isHardened;
```

to remove unsupported kernels.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 122188 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>6 packages marked as broken and skipped:</summary>  <ul>    <li>linuxPackages-libre.rtl8812au</li>    <li>linuxPackages_4_4.rtl8812au</li>    <li>linuxPackages_4_9.rtl8812au</li>    <li>linuxPackages_hardened.rtl8812au</li>    <li>linuxPackages_latest-libre.rtl8812au</li>    <li>linuxPackages_latest_hardened.rtl8812au</li>  </ul></details><details>  <summary>9 packages built:</summary>  <ul>    <li>linuxPackages.rtl8812au</li>    <li>linuxPackages_4_14.rtl8812au</li>    <li>linuxPackages_4_19.rtl8812au</li>    <li>linuxPackages_5_11.rtl8812au</li>    <li>linuxPackages_5_4.rtl8812au</li>    <li>linuxPackages_lqx.rtl8812au</li>    <li>linuxPackages_testing_bcachefs.rtl8812au</li>    <li>linuxPackages_xanmod.rtl8812au</li>    <li>linuxPackages_zen.rtl8812au</li>  </ul></details>

File: pkgs/os-specific/linux/rtl8812au/default.nix [hunk 2]
```diff
@@ -42,9 +42,10 @@ stdenv.mkDerivation rec {
...
+    broken = kernel.kernelOlder "4.10" || kernel.isHardened;
```
Comment (danielfullmer) [inline pkgs/os-specific/linux/rtl8812au/default.nix:49]: Why is this marked broken for kernels older than 4.10?  The github page even says it was built and tested on kernels as old as 5.4.

Comment (jonringer) [reply pkgs/os-specific/linux/rtl8812au/default.nix:49]: 4_4 and 4_9 were failing

Comment (jonringer) [reply pkgs/os-specific/linux/rtl8812au/default.nix:49]: 5.4 is newer than both of those

Comment (danielfullmer) [reply pkgs/os-specific/linux/rtl8812au/default.nix:49]: Whoops. I looked at `4.10` but I read `5.10`.

**Pull request merged**